libraries/chain/controller.cpp

```diff
@@ -646,32 +647,40 @@ struct controller_impl {
-                                                                             majority_permission.id,
-                                                                             active_producers_authority,
-                                                                             conf.genesis.initial_timestamp );
+      if (need_native_accounts) {
+          authority system_auth(conf.genesis.initial_key);
+          create_native_account( config::system_account_name, system_auth, system_auth, true );
+          create_native_account(config::msig_account_name, system_auth, system_auth, true);
+          create_native_account(config::domain_account_name, system_auth, system_auth);
+          create_native_account(config::govern_account_name, system_auth, system_auth, true);
+          create_native_account(config::stake_account_name, system_auth, system_auth, true);
+    
+          auto empty_authority = authority(1, {}, {});
+          auto active_producers_authority = authority(1, {}, {});
+          active_producers_authority.accounts.push_back({{config::system_account_name, config::active_name}, 1});
+    
+          create_native_account( config::null_account_name, empty_authority, empty_authority );
+          create_native_account( config::producers_account_name, empty_authority, active_producers_authority );
+          const auto& active_permission       = authorization.get_permission({config::producers_account_name, config::active_name});
+          const auto& majority_permission     = authorization.create_permission( {},
+                                                                                 config::producers_account_name,
+                                                                                 config::majority_producers_permission_name,
+                                                                                 active_permission.id,
+                                                                                 active_producers_authority,
+                                                                                 conf.genesis.initial_timestamp );
+          const auto& minority_permission     = authorization.create_permission( {},
+                                                                                 config::producers_account_name,
+                                                                                 config::minority_producers_permission_name,
+                                                                                 majority_permission.id,
+                                                                                 active_producers_authority,
+                                                                                 conf.genesis.initial_timestamp );
+      } else {
+          chaindb.emplace<account_object>(config::system_account_name, [&](auto& a) {
```

Comment:
note: `resource_usage_object` and `account_sequence_object` will not be created in genesis for existing account

Reply:
Indeed, there is a confusing moment. In genesis-info.json `cyber` account described without `update` flag, so `create-genesis` create account_object, resource_usage_object and account_sequence_object. In another side, genesis code in `nodeos` check that account exists and update it. Therefore all these object present in DB.
```
> db.resusage.find().limit(1).pretty()
{
	"_id" : ObjectId("5d0a2e2f2abe5100011812c4"),
	"owner" : "cyber",
	"accumulators" : [
		{
			"last_ordinal" : NumberDecimal("0"),
			"value_ex" : NumberDecimal("0"),
			"consumed" : NumberDecimal("0")
		},
		{
			"last_ordinal" : NumberDecimal("0"),
			"value_ex" : NumberDecimal("0"),
			"consumed" : NumberDecimal("0")
		},
		{
			"last_ordinal" : NumberDecimal("0"),
			"value_ex" : NumberDecimal("0"),
			"consumed" : NumberDecimal("0")
		},
		{
			"last_ordinal" : NumberDecimal("0"),
			"value_ex" : NumberDecimal("0"),
			"consumed" : NumberDecimal("0")
		}
	],
	"_SERVICE_" : {
		"scope" : "",
		"rev" : NumberLong(1),
		"payer" : "cyber",
		"size" : 796,
		"ram" : true
	}
}
> db.a
db.abihash        db.account        db.accountseq     db.adminCommand(  db.aggregate(     db.auth(
> db.accountseq.find().limit(1).pretty()
{
	"_id" : ObjectId("5d0a2e2f2abe5100011711ff"),
	"name" : "cyber",
	"recv_sequence" : NumberDecimal("1"),
	"auth_sequence" : NumberDecimal("1"),
	"code_sequence" : NumberDecimal("0"),
	"abi_sequence" : NumberDecimal("0"),
	"_SERVICE_" : {
		"scope" : "",
		"rev" : NumberLong(3),
		"payer" : "cyber",
		"size" : 460,
		"ram" : true
	}
}
```

Reply:
ok